### DISCUSSION TITLE
Laravel Vite cannot be used to bundle a custom theme stylesheet

### DISCUSSION THREAD

**Discussion by realslimsutton** [filamentphp/filament#3018] [Help]
### Package

filament/filament

### Package Version

v2.13.14

### Laravel Version

v9.19

### PHP Version

8.1.4

### Bug description

The Vite directive outputs the entire HTML string instead of just the resource URL. This means when using a custom theme stylesheet as outlined in the documentation [here](https://filamentphp.com/docs/2.x/admin/appearance#building-themes), we're unable to use themes bundled by Laravel Vite as Filament is expecting just the URL.

This could now be more of a problem now that [new Laravel installations come pre-installed with Laravel Vite](https://laravel.com/docs/9.x/vite) instead of Laravel Mix.

One method to get around this issue for now is by publishing the base layout view, and replacing the default theme stylesheet in the head with the "@.vite" directive.

Example:
![Filament-Theme-Vite](https://user-images.githubusercontent.com/87341175/177018725-b03463c0-ee20-44c5-bdf7-ddd30e672ed2.png)

### Steps to reproduce

1. Ensure you're using Laravel Vite to compile a custom theme stylesheet
2. Add the following code to your AppServiceProvider::boot

```
use Illuminate\Foundation\Vite;

Filament::serving(function() {
    Filament::registerTheme(app(Vite::class)(['resources/css/app.css']));
});
```

**Comment by danharrin** [comment]
I want to add official support soon, but does #3004 work?

**Comment by vv-ac** [comment]
Hi, simply for one's own planning with setting up sth new. Is there an approx. deadline tied to Vite support?

Have a nice evening!

**Reply by danharrin** [reply]
I'll implement it when its stable... atm there are various issues with it in TALL stack applications and in popular environments, even Valet.

**Reply by vv-ac** [reply]
Thanks, we gonna wait for them then ;-).
